### PR TITLE
Add manufacturability report + update antenna check/repair steps

### LINKED ISSUE #346: Add manufacturability report
### Description

Add manufacturability report similar to the one in OpenLane 1

### Proposal

Add a new step that print a manufacturability report.

### PR DESCRIPTION
**Pull request by kareefardi**
* `Classic`
  * `OpenROAD.CheckAntennas` now run twice: once after `OpenROAD.GlobalRouting` and once after `OpenROAD.DetailedRouting`
* `OpenROAD.GlobalRouting`
  * **API Break**: Antenna check is no longer run and relevant metric is no longer updated.
* Added `Misc.ReportManufacturability` which displays results of LVS, DRC, and antenna checks.
* `OpenROAD.RepairAntennas`
  * Transformed into a composite step of a private diode insertion step and `OpenROAD.CheckAntennas`.
* `OpenROAD.CheckAntennas`
  * Moved summary to `reports` directory within step directory for consistency across signoff steps.
* `Netgen.LVS` 
  * Moved summary to `reports` directory within step directory for consistency across signoff steps.
* `openlane.steps.Step`:
  * Log files printed to interactive terminals are now clickable links. 
* Updated OpenROAD to `c1fc5f7`
  * Updated OpenSTA to `42b994d`

---

Fixes #346

---

Dev notes:
After changing `OpenROAD.CheckAntennas`, there was a crash. Turns out there was code duplication of antenna reporting in `grt.tcl` and `antenna_check.tcl` so I changed GlobalRouting step to a compositestep. This maintains backwards compatibility and avoids code duplication. Perhaps the naming of the new underlying step is a bit awkward.
Edit:
We opted for not having it as composed step and adding antenna check later to the classical flow when the bug is fixed in OpenROAD.
Edit: Bug fixed and merged into this PR.

### CODE REVIEW

File: openlane/steps/openroad.py [hunk 1]
```diff
@@ -1183,23 +1183,39 @@ def __lt__(self, other):
...
+            open(report_path)
         )
+        self.__summarize_antenna_report(report_path, report_summary_path)
 
         return views_updates, metrics_updates
 
 
 @Step.factory.register()
-class GlobalRouting(CheckAntennas):
+class GlobalRouting(OpenROADStep):
```
Comment (donn) [inline openlane/steps/openroad.py:1218]: Not sure why. The step does mention that it performs the antenna check in its docstring and it doesn't hurt.

Comment (kareefardi) [reply openlane/steps/openroad.py]: I have changed it to a CompositeStep. A more openlane2 way of doing things.

**Comment by kareefardi** [comment]
CI failed due to this issue https://github.com/The-OpenROAD-Project/OpenROAD/issues/4560

**Comment by donn** [comment]
Tbh, I think the change to GlobalRouting is kind of unnecessary personally. I understand it's a more "OpenLane 2" way of doing things, but also, realistically it works just fine and GlobalRouting having a built-in antenna check doesn't hurt THAT much. Comparatively, there is fallout from this and the name "GlobalRoutingOnly" itself is a bit unorthodox- if you're insistent, I would just break the API for `GlobalRouting` and call it a day.

**Comment by kareefardi** [comment]
> Tbh, I think the change to GlobalRouting is kind of unnecessary personally. I understand it's a more "OpenLane 2" way of doing things, but also, realistically it works just fine and GlobalRouting having a built-in antenna check doesn't hurt THAT much. Comparatively, there is fallout from this and the name "GlobalRoutingOnly" itself is a bit unorthodox- if you're insistent, I would just break the API for `GlobalRouting` and call it a day.

It works just fine but there is duplication and there is opportunity here to remove it. I am not opposed to breaking the API but we still need an OpenROAD fix to be able to run antenna check after global routing.

**Review by donn — APPROVED** [review]
Went ahead and merged OpenROAD + did some changes for consistency.

Ball's in your court. I'll approve, you may re-review and merge.

**Pull request merged**